src/nvim/window.c

```diff
@@ -6988,3 +6993,15 @@ win_T *lastwin_nofloating(void) {
   }
   return res;
 }
+
+/*
+ * check if there is at least one unanchored floating window.
+ */
+bool check_unanchored_floats(void) {
+  FOR_ALL_TAB_WINDOWS(tp, wp) {
```

Comment:
this doesn't seem right. The relevant condition should be unanchored floats in the current tabpage.

Reply:
As mentioned in the other comment, I made it like this so that it could be called from the `win_close` function, to keep all the checks together in a single function, so that `win_close` could return `FAIL` for all such cases when window(s) weren't closed.

Reply:
But it doesn't make any sense. We should not stop the last window of tabpage A (current) from being closed because there are floats on tabpage B.

Reply:
I did think of it but I don't know why I thought that firstwin and lastwin_nofloating() kept the first and last windows for the current tab. I guess I shouldn't skip on sleep as much.